basic_algs/src/main.rs

```diff
@@ -0,0 +1,66 @@
+fn min(array: &[f64; 10]) -> f64 {
+    let mut cur_min = array[0];
+    for element in array {
+        if cur_min > *element {
+            cur_min = *element
+        }
+    }
+    return cur_min
+}
+
+fn is_prime(n: &i32) -> bool {
+    if *n == 1 {
+        return false
+    }
+    if *n == 2 {
+        return true
+    }
+    for i in 2..*n {
+        if *n % i == 0 {
+            return false
+        }
+    }
+    return true
+}
+
+fn nth_prime(n: &i32) -> i32 {
+    let mut cur_num = 1;
+    let mut i = 0;
+    while i != *n {
+        cur_num += 1;
+        if is_prime(&cur_num) {
+            i+=1;
+        }
+    }
+    return cur_num
+}
+
+fn bin_search(array: &[f64; 10], key: &f64) -> i32 {
+    let mut l: i32 = -1;
+    let mut r: i32 = 10;
+    while l < r - 1 {
+        let m: usize = ((l + r) / 2) as usize;
```

Comment:
На этой строчке можно не указывать тип переменной, так как используется `as`:
```suggestion
        let m = ((l + r) / 2) as usize;
```